packages/connect/src/types/settings.ts

```diff
@@ -65,15 +65,21 @@ export interface ConnectSettingsWeb {
 export interface ConnectSettingsWebextension {
     /** _extendWebextensionLifetime features makes the service worker in @trezor/connect-webextension stay alive longer */
     _extendWebextensionLifetime?: boolean;
+    coreMode?: 'auto' | 'popup' | 'suite-desktop';
 }
 export interface ConnectSettingsMobile {
     deeplinkUrl: string;
     deeplinkOpen?: (url: string) => void;
     deeplinkCallbackUrl?: string;
+    coreMode?: 'deeplink';
 }
 
 export type ConnectSettings = ConnectSettingsPublic &
     ConnectSettingsInternal &
-    ConnectSettingsWeb &
-    ConnectSettingsWebextension &
-    ConnectSettingsMobile;
+    // coreMode is a common parameter between these, so it is explicitly handled here for correct handling
```

Comment:
hmm I couldn't find a better way how to get rid of this nastiness quickly. it would probably require introducing some discriminator field each settings group and implement runtime checks where needed I guess